### PR TITLE
Add Faqwiki.us source (beta branch)

### PR DESCRIPTION
**Pull request by bregassa**
Add faqwiki.us source for beta branch

Open Issue :
https://github.com/LNReader/lnreader-sources/issues/789
https://github.com/LNReader/lnreader-sources/issues/619

main branch PR:
https://github.com/LNReader/lnreader-sources/pull/791#issue-2142651712
https://github.com/LNReader/lnreader-sources/pull/791#issuecomment-1953582463

### CODE REVIEW

File: plugins/english/faqwikius.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  async popularNovels(
+    page: number,
+    { showLatestNovels }: Plugin.PopularNovelsOptions,
+  ): Promise<Plugin.NovelItem[]> {
+    const sort = showLatestNovels
+      ? ""
+      : "";
+
+    const body = await fetchApi(this.site + sort).then((res) => res.text());
```
Comment (nyagami) [inline plugins/english/faqwikius.ts:21]:
```suggestion
    const body = await fetchApi(this.site).then((res) => res.text());
```

File: plugins/english/faqwikius.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+   // Extract text from individual paragraphs and join 
+    const chapterText = chapterParagraphs.map((index, element) => {
+        return loadedCheerio(element).text().trim();
+    }).get().join('\n\n'); // Add newlines between paragraphs
+
+    return chapterText;
+  }
+
+  async searchNovels(searchTerm: string): Promise<Plugin.NovelItem[]> {
+    const body = await fetchApi(this.site).then((res) => res.text());
```
[the remark this answers is not in the document]
Comment (bregassa) [reply plugins/english/faqwikius.ts:93]: > image this site has search tool. Could you implement it?

Using the search tool on that site will output posts without an image cover + redundant posts. The site homepage lists all the novels they have, so using a filter on the main page for search is better than actually using their search bar.

**Review by nyagami — APPROVED** [review]

File: plugins/english/faqwikius.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      .map((index, element) => {
+        const name = loadedCheerio(element).find("strong").first().text();
+        let cover = loadedCheerio(element).find("img").attr("data-ezsrc"); 
+
+        // Remove the appended query string 
+        if (cover) {
+             cover = cover.replace(/\?ezimgfmt=.*$/, ''); // Regular expression magic!
+        }
+
+        const path = "" + loadedCheerio(element).find("a").attr("href");
```
Comment (nyagami) [inline plugins/english/faqwikius.ts:34]: image
Pls remove the domain in novel's path

Comment (bregassa) [reply plugins/english/faqwikius.ts]: Done https://github.com/LNReader/lnreader-sources/pull/792/commits/6fb6a355b3b28964e25b3991819a922d006b322e

**Comment by nyagami** [comment]
image
still absolute path in searchNovels

**Comment by bregassa** [comment]
> image still absolute path in searchNovels

my bad, fixed.

**Pull request merged**